Fri May  2 06:57:43 BST 2014  Adam Vogt <vogt.adam@gmail.com>
  * derive Applicative instance for Query
diff -rN -u old-xmonad/XMonad/Core.hs new-xmonad/XMonad/Core.hs
--- old-xmonad/XMonad/Core.hs	2015-01-08 21:40:45.320979242 +0000
+++ new-xmonad/XMonad/Core.hs	2015-01-08 21:40:45.367979193 +0000
@@ -155,7 +155,7 @@
 
 type ManageHook = Query (Endo WindowSet)
 newtype Query a = Query (ReaderT Window X a)
-    deriving (Functor, Monad, MonadReader Window, MonadIO)
+    deriving (Functor, Applicative, Monad, MonadReader Window, MonadIO)
 
 runQuery :: Query a -> Window -> X a
 runQuery (Query m) w = runReaderT m w
